gm4_boat_chests/data/gm4_boat_chests/functions/merge.mcfunction

```diff
@@ -0,0 +1,12 @@
+
+advancement grant @a[distance=..5] only gm4:boat_chests
+
+tag @s add gm4_bc_old_boat
+function gm4_boat_chests:old_boat
+
+execute as @e[type=chest_minecart,distance=..3,sort=nearest,limit=1,tag=!gm4_bc_chest] run function gm4_boat_chests:old_minecart
```

Comment:
```suggestion
execute as @e[type=chest_minecart,sort=nearest,distance=..2,predicate=!gm4_boat_chests:is_passenger,limit=1] run function gm4_boat_chests:old_minecart
 ```
 For consistency purposes, use `distance=..2` just like your main function's `if entity` check. Additionally, see the reasoning for my suggested change to the `main` function for why `tag=!gm4_bc_chest` is replaced with `predicate=!gm4_boat_chests:is_passenger`.